### PR TITLE
Enable fast Huffman & Huffman zig-zag transform for Arm Neon

### PR DESCRIPTION
**Pull request by Developer-Ecosystem-Engineering**
Implements fast Huffman on macOS, then builds on top of those changes to enable Huffman zig-zag transform

### CODE REVIEW

**Comment by meshula** [comment]
Ah very nice, thank you! I'm wondering if you have any benchmarking results you might be able to report? I'm sure there's an improvement, but I am curious as to how much of a difference it might be?

**Review by meshula — APPROVED** [review]

File: src/lib/OpenEXR/ImfFastHuf.cpp [hunk 1]
```diff
@@ -11,6 +11,40 @@
...
 
+// Static enabling/disabling the fast huffman decode
+
+
+#if defined(__clang__)
+//
+// Enabled for clang on Apple platforms (tested):
+//
+
+#    if defined(__APPLE__)
```
Comment (kmilos) [inline src/lib/OpenEXR/ImfFastHuf.cpp:22]: This should also be supported by the [MSYS2 CLANGARM64 environment](https://www.msys2.org/docs/environments/), so could in theory change to just `__aarch64__`?

**Review by kmilos — COMMENTED** [review]
Aarch64 support could actually be expanded to even more platforms & compilers potentially: Clang for MinGW, GCC on *nix, and MSVC on WoA.

Though it's fine if you just want to keep to the tested ones ATM.

File: src/lib/OpenEXR/ImfFastHuf.cpp (hunk 1 above)
Comment (Developer-Ecosystem-Engineering) [reply src/lib/OpenEXR/ImfFastHuf.cpp:22]: We would not enable support on configurations we have not tested, in theory it could be done!

**Comment by Developer-Ecosystem-Engineering** [comment]
> Ah very nice, thank you! I'm wondering if you have any benchmarking results you might be able to report? I'm sure there's an improvement, but I am curious as to how much of a difference it might be?

We did see a significant improvement in our own testing. If there are public tests the project would like to see, we are happy to run and provide

**Comment by cary-ilm** [comment]
Thanks for the contribution!  I added "Arm Neon" to the PR title, for clarity.

**Comment by mandree** [comment]
This causes build failures on ARMv7, see #1367

**Pull request merged**